.github/workflows/test.yml

```diff
@@ -1,21 +1,48 @@
 name: test
-on: [push, pull_request]
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
 
 permissions:
   contents: read
 
+env:
+  TESTFLAGS: -v
+  BUILDFLAGS: -v
+
 jobs:
-  test:
+  build:
     strategy:
       matrix:
         go-version: [1.20.x, 1.21.x]
         os: [ubuntu-latest, windows-latest, macos-latest]
     runs-on: ${{ matrix.os }}
-    timeout-minutes: 10
```

Comment:
can you keep a timeout? Github Action's default is 6 hours, and I definitely have had cases where something "stuck" and kept running for that time.

Reply:
Can do.